java/src/test/java/io/cucumber/gherkin/utils/pretty/PrettyTest.java

```diff
@@ -316,6 +316,120 @@ public void commentAtStartAndEndOfFile() {
                 prettyPrint(gherkinDocument, Syntax.gherkin));
     }
 
+    @Test
+    void commentBeforeExamples() {
+        GherkinDocument gherkinDocument = parse("Feature: Comment before an Examples\n" +
+                "\n" +
+                "  Scenario Outline: with examples\n" +
+                "    Given the <value> minimalism\n" +
+                "    # then something happens\n" +
+                "\n" +
+                "    Examples:\n" +
+                "    | value |\n" +
+                "    | 1     |\n" +
+                "    | 2     |\n");
+        assertEquals("Feature: Comment before an Examples\n" +
+                        "\n" +
+                        "  Scenario Outline: with examples\n" +
+                        "    Given the <value> minimalism\n" +
+                        "\n" +
+                        "    # then something happens\n" +
+                        "    Examples:\n" +
+                        "      | value |\n" +
+                        "      | 1     |\n" +
+                        "      | 2     |\n",
+                prettyPrint(gherkinDocument, Syntax.gherkin));
+    }
+
+    // This demonstrate some limitations like:
+    // - Comments between a `@tag` and a `Scenario:` are pushed before the @tag
+    // - Dangling comments (e.g. with intermediate EOL) are compacted next to some block
+    // - Contiguous `###[...]` are turned into `# ##[...]`
+    // - Comments after an moreIndented block are moved as indent of the following lessIndented block
+    @Test
+    void commentsBeforeAndAfterKeywords(){
+        GherkinDocument gherkinDocument = parse("# before Feature\n" +
+                "Feature: Comments everywhere\n" +
```

Comment:
I wonder if it would be useful to harmonize our test cases between languages. How this might look:
1) new directory `testdata/pretty/before` and `testdata/pretty/after`
2) @blacelle move our examples into `before/foo.feature` and `after/foo.feature`
3) we read these files from disk in our JS/Java test code, and both suites execute all examples

The benefit is, we ensure that both languages have identical treatment of comments when prettying. I am not sure how important this behavior is.

If interested, I can get this started on #16.

Reply:
> The benefit is, we ensure that both languages have identical treatment of comments when prettying. I am not sure how important this behavior is.

This would definitely be a good thing.

---

I would be happy to follow this structure, preferable in a separate PR. But I do not get how to contribute to this repository as this PR seems stuck :S.